addons/account/models/account_reconcile_model.py

```diff
@@ -74,6 +72,8 @@ class AccountReconcileModelLine(models.Model):
     * Fixed: The fixed value of the writeoff. The amount will count as a debit if it is negative, as a credit if it is positive.
     * From Label: There is no need for regex delimiter, only the regex is needed. For instance if you want to extract the amount from\nR:9672938 10/07 AX 9415126318 T:5L:NA BRT: 3358,07 C:\nYou could enter\nBRT: ([\\d,]+)""")
     tax_ids = fields.Many2many('account.tax', string='Taxes', ondelete='restrict', check_company=True)
+    # Relational fields for applying domain in the view
+    tax_ids_domain = fields.Binary(related="model_id.tax_ids_domain", string="Tax ids domain")
```

Comment:
You don't really need this, you could just have a char field getting the `type_tax_use` instead, and use `('type_tax_use', '=?', type_tax_use)` as a string in the domain of `tax_ids` instead. (note that you can also use `if ... else`and stuff like that in a string domain expression evaluated by the client)
The domain on the company is already taken into account thanks to `check_company=True`